Moonshot/player/player.gd

```diff
@@ -177,23 +177,25 @@ func take_damage(damage, attack_dir, is_damage_tile: bool = false) -> void:
 			attack_dir = sign(get_global_position().x - attack_dir.x)
 			stagger_player(attack_dir,is_damage_tile)
 
-	if Utils.player_stats.health <= 0:
-		on_death()
-		
-	Utils.player_stats.take_damage(1)
 
-func add_health(health):
+func add_health(health) -> void:
 	var new_health = Utils.player_stats.health + health
 	Utils.player_stats.health = min(Utils.player_stats.max_health, new_health)
 
 
 func on_death() -> void:
 	CombatSignalController.emit_signal("player_kill")
 	_on_Player_health_depleted()
+	CombatSignalController.disconnect("damage_player", self, "take_damage")
```

Comment:
Disconnect the signal so `take_damage()` will not be called again until game restart. The signal is connected in `_ready()`.